### DISCUSSION TITLE
Attributt: helsepersonellets funksjonelle rolle

### DISCUSSION THREAD

**Discussion by Helgebjert** [NorskHelsenett/Tillitsrammeverk#4] [Q&A]
Hei 
Jeg vet ikke om dette blir riktig men jeg prøver meg på en diskusjon:) Si gjerne i fra om dette er riktig måte å gjøre det på. 
 
Hvorfor er helsepersonellets funksjonelle rolle satt til "obligatorisk = JA"? ref. kap 3.2.6 

I beslutningsunderlaget som ble tatt frem før jul ble det anbefalt til å ikke være en del av informasjonen som overføres i sikkerhetsbeviset. basert på følgende:  "_Etter det vi erfarer så langt er denne informasjonen i dag ikke harmonisert på egnet måte. Informasjonselementene som beskriver ansattrolle og funksjonell rolle er ikke påkrevd brukt i journalsystemer i dag, og vi vet ikke om denne informasjonen finnes."

Hilsen 
Helge :)

**Comment by steinarnoem** [comment]
Hei Helge, det er kjempefint at du bidrar!

Vi har harmonisert datamodellen i vår spesifikasjon med datamodellen i EHDSI hvor det er angitt at verdiene for dette attributtet skal komme fra STYRK-08 (ISCO-08) og er obligatorisk. STYRK-08 er brukt i alle virksomheter i Norge i dag, også innenfor helse- og omsorg.

Min mening er at rollene som er definert i STYRK-08 er veldig oversiktlige, og at dette bør være enkelt å implementere i journalsystemer som har tilgangsstyring. Derfor mener jeg at argumentet i beslutningsgrunnlaget ikke bør følges. 

Konklusjonen i arbeidsgruppa var vel uansett også at at detaljering av datamodellen skulle gjøres i gjennomføringsfasen?

Kodeverket STYRK-08 finner du her:
https://www.ssb.no/klass/klassifikasjoner/7

I STYRK-08 er det definert rundt 30 roller som er relevant for helsepersonell i helse- og omsorgssektoren, hvorav bare et subsett vil være relevant for de ulike helsetjenestetypene der helsepersonellet jobber.

S

**Reply by Helgebjert** [reply]
Takk for svar! 
Jeg ser at Eirik tar opp det sammen + noen til under "issues".  Noe å forfølge er om dette er informasjon som er tilgjengelig I EPJ. Diskusjonen kan lukkes her og utforskes nærmere med Oslo Kommune. 

mvh, Helge

**Reply by eirikbroen** [reply]
Som @steinarnoem skriver over er kodeverk her er satt til å være STYRK-08, informasjonskilde er EPJ/HR. Antar at mange virksomheter har EPJ uten direkte integrasjon med HR system og dermed enten må lage en slik integrasjon eller gjøre en form for dobbeltføring. Burde vi foreslå en mulig mapping fra funksjonelle roller i EPJ? Typ - EPJ Rolle "Lege" i et system i primærhelsetjenesten = STYRK-08 2211 - Allmennpraktiserende Leger?

**Reply by stvgr** [reply]
I Stavanger kommune har vi ikke direkte integrasjon mellom EPJ/HR i forhold til helsepersonellets roller. I HR systemet ser det ut til at det er utdanningskode som er registrert på meg og det er etter det jeg kan se Styrk-98. Jeg ser at dette antakelig blir gjort om til STYRK08 kode hos SSB, så det kan være det vil endre seg i HR systemet også.
Mapping høres ut som en god ide.

**Reply by Helgebjert** [reply]
Ref.  på denne diskusjonen: 

_I beslutningsunderlaget som ble tatt frem før jul ble det anbefalt til å ikke være en del av informasjonen som overføres i sikkerhetsbeviset. basert på følgende: "_Etter det vi erfarer så langt er denne informasjonen i dag ikke harmonisert på egnet måte. Informasjonselementene som beskriver **ansattrolle og funksjonell rolle er ikke påkrevd brukt i journalsystemer i dag**, og vi vet ikke om denne informasjonen finnes."_

Slik jeg forstår både Stavanger og Oslo kan Styrk-08 være et aktuelt kodeverk, men at dette ikke er informasjon som er tilgjengelig eller benyttes i EPJ.  Da stemmer fortsatt det vi skrev i beslutningsunderlaget. Mitt forslag er derfor at dette informasjonselementet endres fra "obligatorisk=JA" til "obligatorisk=NEI".

**Reply by stvgr** [reply]
Jeg er enig i at det er OK hvis det blir endret til obligatorisk=NEI.

**Comment by steinarnoem** [comment]
Jeg har lagt inn noen relevante kodeverk som gir en liten ide om overlapp.. 
Ta en titt på https://github.com/NorskHelsenett/Tillitsrammeverk/blob/main/specs/misc/sammenligning_av_funksjonell_rolle.md

Slik vi har lagt opp spesifikasjonene kan egentlig alle disse kodeverkene uttrykket i JSON datastruktur og i SAML tokens ved at vi lar konsumenten uttrykke en kode og referere til kodeverket/listen over gyldige verdier.

**Reply by thomiz** [reply]
Her har dere vel blandet sammen rolle med kvalifikasjon. HL7 FHIR skiller for eksempel på disse begrepene. Det er vel også noen andre kodeverk fra Volven som kan være relevante her, slik det er uttrykt i HL7 Norge sin no-basis profil for [PractitionerRole](https://simplifier.net/hl7norwayno-basis/nobasispractitionerrole). (Men husk at no-basis bare er et minimumssett i forhold til selve samhandlingen og ikke ivaretar alle behov knyttet til en mer generell tillitsmodell).

**I no basis endte vi på [Helsepersonells funksjoner mv. (OID=9034)](https://volven.no/produkt.asp?id=494103&catID=3&subID=8)**

Når det gjelder kvalifikasjon valgte vi i no-basis arbeidet å foreslå tre kodeverk fra volven for bruk i [Pracitioner.qualification](https://simplifier.net/hl7norwayno-basis/nobasispractitioner)
* [Autorisasjonskodeverket (9060)](https://volven.no/produkt.asp?id=494115&catID=3&subID=8)
* [Utdanningsnivå Helsepersonellregisteret (OID=7703)](https://volven.no/produkt.asp?id=495082&catID=3&subID=8)
* [Helsepersonellregisterets (HPR) klassifikasjon av spesialiteter (OID=7426)](https://volven.no/produkt.asp?id=495071&catID=3&subID=8)

I HL7 Norge diskuterte vi ganske lenge hva som skal være med i basisprofilen på dette området og endte opp med kodeverk som da var i utstrakt bruk på området, knyttet til utveksling av informasjon. Hvis HL7 Norge skulle utformet en modell for å uttrykke et tillitsrammeverk som var mer generelt og mer implementasjonsnært skulle nok flere verdisett blitt med.

**Reply by steinarnoem** [reply]
Det er veldig godt å se at du kan bidra i diskusjonen @thomiz!

Jeg kjente ikke til 9034 - så det var nyttig informasjon. Hva synes du om EHDSI sitt valg av ISCO-08 som kodeverk for å uttrykke funksjonell rolle?

Jeg fjerner dokumentet hvor jeg gjorde et forsøk på å mappe, men kunne godt ha tenkt meg å høre mer om hva du (og andre i HL7) mener er et velegnet kodeverk for å uttrykke funksjonell rolle..

**Reply by thomiz** [reply]
Jeg tror det er helt avhengig hvordan man definerer begrepet funksjonell rolle til å bety. I arbeidet med basisprofilene for Practitioner og PractitionerRole la vi jo tolkningen i kodeverk 9034 til grunn, altså funksjoner som helsepersonellet utfører overfor pasient, og egentlig uavhengig av kvalifikasjoner og yrkestitler. Men det eksempelet som lå i HL7 FHIR standarden i utgangspunktet for PractitionerRole.code ligner jo veldig på ISCO-08, altså yrkestitler. Så her er det kanskje grunnlag for en review av det arbeidet som ble gjort i HL7 Norge med disse basisprofilene også?

**Reply by eirikbroen** [reply]
Veldig interessant diskusjon og fint bidrag fra deg @thomiz. Jeg kjente heller ikke til 9034, men ser jo at verdiene i dette kodeverket godt beskriver en funksjonell rolle. Listen er kanskje litt kort for å favne alle aktuelle roller, men det kan eventuelt utvides. Veldig usikker på i hvilken grad ulike EPJ systemer har et forhold til 9034 og dermed om det er praktisk mulig å ta dette i bruk på kort sikt. Vet vi noe om utbredelse og bruk? 

Det jeg liker godt med dette kodeverket er at det gir større fleksibilitet mht den rollen helsepersonellet spiller i et enkelt behandlingsforløp. Er vel for eksempel slik at en ansatt lege med Styrk-08 kode _2211 - Allmennpraktiserende leger_ gjerne veksler mellom å være både _3 - Behandlingsansvarlig lege_, _13 - Innleggende lege_ og _10 - Utskrivende lege_

**Reply by thomiz** [reply]
Jeg er litt usikker på den faktiske bruken av kodeverket. Den er en del av [Hodemeldingen](https://git.sarepta.ehelse.no/publisert/standarder/raw/master/kravdokument/hodemelding/HIS_80601_2006-Hodemelding%20-oppdatert.pdf) som jo er svært utbredt og i faktisk bruk i hele helsesektoren i tilknytning til ulike meldingsstandarder. Men elementet som bruker 9034 er ikke obligatorisk, noe som kan føre til at det som faktisk overføres ikke nødvendigvis inneholder denne informasjonen og mange systemer kan velge å ikke implementere dette støtte for akkurat dette kodeverket. Vi må kanskje høre med meldingsgjengen i NHN/eHelse om de vet noe om status her?

**Reply by stvgr** [reply]
9034 er jo også veldig lite kommunerettet slik kodeverdiene er definert nå.

**Reply by AMJohnsen** [reply]
Hvis 9034 skal benyttes må det utvides og ryddes i betraktelig. Som @stvgr sier så er det lite kommunerettet. Noen av kodeverdiene kan tolkes forskjellig. Kan fort ende med at alle andre enn klinikere, jordmødre osv. faller inn kodeverdiene informasjonsansvarlig, kontakt hos mottar osv., som gir liten verdi.

**Reply by OK-SverreM** [reply]
Jeg ser at et innlegg jeg tidligere har skrevet i en [annen diskusjon](https://github.com/NorskHelsenett/Tillitsrammeverk/discussions/24) er ganske relevant for denne tråden, så jeg kopierer det inn her:

I tidligere runder med både HSØ og PDS har det vært snakk om tre distinkte typer roller:

- **Autorisasjonsrolle** (OID 9060). Denne er grei, hentes fra HPR.
- **Ansattrolle**. I betydningen "stilling du er ansatt i". For denne type mer administrativ informasjon, vil primærkilden være HR-systemet (ikke EPJ). Her ser jeg det kan gi mening å benytte STYRK kodeverk, som vel nettopp benyttes for innrapportering fra HR-system til arbeidsgiver/arbeidstaker-registeret (via EDAG-ordningen). Men: Vår konklusjon så langt er at denne informasjonen gir begrenset / ingen merverdi utover autorisasjonsrollen, + at STYRK-koder ikke finnes i EPJ-systemene (det er ikke deres naturlige habitat).
- **Funksjonell rolle**. HSØ har gjort et omfattende arbeid med å etablere et internt system for dette (se https://github.com/NorskHelsenett/Tillitsrammeverk/blob/main/specs/misc/sammenligning_av_funksjonell_rolle.md). Men det finnes ikke noe tilsvarende for kommunene f.eks.


Jeg tror vi gjør klokt i å ikke blande disse ulike typene roller sammen med hverandre.
Hvis vi skal ha et rollebegrep i tillegg til Autorisasjonsrolle, så bør det være veldig klart om vi snakker om Ansattrolle (hvor STYRK er aktuelt som kodeverk) eller det mer helsetjenestespesifikke Funksjonell rolle (hvor det per nå ikke finnes ? / er pekt på noe felles kodeverk).

Konklusjon på kort sikt (dvs. 2023 / realiseringen av behovskøelement # 27)?
Hvis man skal ta med **Ansattrolle** og/eller **Funksjonell rolle** i Tillitsrammeverket, så kan ikke disse per nå settes som obligatoriske.
I tillegg må det være helt tydelig hvilken av de to rolletypene man tar med, og for hvilket formål. Det tjenstlige behovet (som vel skal være det viktigste grunnlaget for tilgang i Tillitsmodellen) belyses neppe så mye bedre ved å ha med en rekke forskjellige typer roller?

**Reply by Helgebjert** [reply]
@OK-SverreM  
enig. Ttilsvarende kommentar fra meg ligg lengre opp i tråden. "Mitt forslag er derfor at dette informasjonselementet endres fra "obligatorisk=JA" til "obligatorisk=NEI"." I den første runden vi skal prøve ut nå bør vi heller starte med minimum. Dersom man ser at dette er et stort behov og noe som mangler kan man heller legge til. Generelt er det vanskeligere å ta noe bort.

**Reply by richardhus** [reply]
Jeg støtter også konklusjonene over, fra Helse Sør-Øst sin side ønsker vi muligheten for å kunne formidle en funksjonell rolle som i størst mulig grad, i alle fall på sikt, beskriver helsepersonellets ansvar og oppgaver i forhold til pasientbehandlingen. Da gir det informasjon som gir grunnlag for å vurdere hva som er relevante og nødvendige opplysninger for vedkommende i forhold til kolleger med tilsvarende stillingstittel og/eller utdanning/autorisasjoner. Det er også høyst relevant for personell uten formell autorisasjon i helsepersonellregisteret. Men jeg tror det bør være mulig å formidle verdier fra ulike kodeverk, og langt mer granulert enn det som ligger i STYRK i alle fall, selv om det også kan være et aktuelt nivå å formidle som noe annet enn "funksjonell rolle", for STYRK er ikke spesielt funksjonelt orientert slik jeg ser det. Volven 9034 derimot er et langt bedre utgangspunkt å bygge videre på, men jeg tviler på at vi enkelt kan avgi det pr. nå, men det er det bedre om @troelde uttaler seg om på vegne av DIPS-kunder :)

I første omgang ønsker vi fra Helse Sør-Øst å kunne videreføre vår bruk av functional_role-attributtet slik at roller fra vår regionale rolleliste (som @OK-SverreM viste til lenger opp) kan formidles inntil vi eventuelt konkluderer på noe mer standardisert nasjonalt.

**Reply by micer2004** [reply]
Må bare spørre, kom på et scenario som jeg ikke vet om den er aktuelt eller ikke. 

Finnes det helsepersonell som har en type av godkjenning (f.eks. **HPR-autorisasjon** som **apotektekniker**) men i praksis jobber som en annen type helsepersonell (f.eks. **funksjonell rolle** som **sykepleier** i kommunen). Dvs. at autorisasjon fra HPR-registeret samsvarer ikke med rollen i virksomheten som hen er ansatt i, til tross at vedkommende er godkjent helsepersonell. Kanskje et sjeldent scenario men det kan poengtere at bruk av HPR-registeret kan være misvisende, dog det beste vi har, siden HPR-registeret dokumenterer godkjent utdanning av helsepersonellet, ikke rollen i helsevesenet.

**Reply by richardhus** [reply]
@micer2004: Jeg tenker at helsepersonell som for eksempel leger og sykepleiere fint kan ha roller i lokalt EPJ uten at de nødvendigvis har hjemmel for bruk av kjernejournal når de opptrer i roller som forsker eller personalleder og at det kanskje er et vel så relevant eksempel på at HPR-roller ikke nødvendigvis sier den fulle sannheten om hva man kan få tilgang til. Vi kjenner vel alle helsepersonell som nå jobber i IKT, som konsulenter/rådgivere eller liknende og som ikke lenger driver pasientbehandling i den rollen. Men det er nok mest aktuelt når de også har tilgang til kliniske systemer og kan initiere API-kall eller uthopp til for eksempel kjernejournal.
Jeg ser ikke bort fra at det kan finnes eksempler slik du skisserer også altså.

**Reply by troelde** [reply]
Svar til @richardhus 
> Men jeg tror det bør være mulig å formidle verdier fra ulike kodeverk, og langt mer granulert enn det som ligger i STYRK i alle fall, selv om det også kan være et aktuelt nivå å formidle som noe annet enn "funksjonell rolle", for STYRK er ikke spesielt funksjonelt orientert slik jeg ser det. Volven 9034 derimot er et langt bedre utgangspunkt å bygge videre på, men jeg tviler på at vi enkelt kan avgi det pr. nå, men det er det bedre om @troelde uttaler seg om på vegne av DIPS-kunder :)

I DIPS kan kundene selv definere innholdet i kodeliste som skal brukes for _funksjonell rolle_ (det litt misvisende begrepet "stillingstype" på en brukerrolle), men det er kun støtte for **ett** kodeverk. DIPS har p.t. ingen metadata om hvilket kodeverk som kodelisten representerer (RHF-spesifikt eller nasjonalt).

**Comment by steinarnoem** [comment]
Feature-teamet som har det redaksjonelle ansvaret for spesifikasjonene hadde et arbeidsmøte 14.03.2023 hvor temaet "funksjonell rolle" ble diskutert. Slik det ser ut nå har vi ikke en felles forståelse av hva funksjonell-rolle beskriver, hvordan det skal benyttes, og hvilke kodeverk som skal benyttes. 

I tillegg ser vi at vår tolkning av "funksjonell rolle" ikke er sammenfallende med hvordan den brukes i EU (EHDS).


Derfor ble det konkludert med følgende:

- "functional_role" tas ut av spesifikasjonen
- "authorization" - som angir anvendt/gjeldende rolle fra HPR tas inn i spesifikasjonen


Møtereferat: https://github.com/NorskHelsenett/Tillitsrammeverk/blob/main/meeting_minutes/2023-03-14%20temam%C3%B8te.md

**Comment by mortsten** [comment]
Legger inn noen betraktninger om funksjonell rolle skal tas inn på et senere tidspunkt i spesifikasjonen.

Rolle i seg selv har jo flere formål og de ulike kodeverkene beskriver ulike formål 
Behovet som det har vært pekt på er loggkontroll, logg til innbygger og tilgangsstyring.

Loggkontroll/logg til innbygger
Funksjonell rolle i HSØ' spesifikasjon av funksjonell rolle #58 sammen med PurposeOfUseLocal gir en beskrivelse av tjenstlig behov.
Funksjonell rolle slik NO-Bassis definerer det,  kan etter beskrivelsen dokumentere tjenstlig behov (roll + purpose_local )men må trolig utvides til logg formål.

Tilgangskontroll/differensiering av tilgang
Kvalifikasjon/Autorisasjons rolle kan i kombinasjon med andre attributter benyttes til å avgrense hva som utleveres.
